providers/encoders.inc

```diff
@@ -11,102 +11,161 @@
-    ENCODER("DSA", "yes", "text", "public", dsa_pub_text_encoder_functions),
+    ENCODER("DSA", "yes", "text", "private",
+            dsa_priv_to_text_encoder_functions),
+    ENCODER("DSA", "yes", "text", "public",
+            dsa_pub_to_text_encoder_functions),
     ENCODER("DSA", "yes", "text", "parameters",
-        dsa_param_text_encoder_functions),
-    ENCODER("DSA", "yes", "der", "private", dsa_priv_der_encoder_functions),
-    ENCODER("DSA", "yes", "der", "public", dsa_pub_der_encoder_functions),
-    ENCODER("DSA", "yes", "der", "parameters", dsa_param_der_encoder_functions),
-    ENCODER("DSA", "yes", "pem", "private", dsa_priv_pem_encoder_functions),
-    ENCODER("DSA", "yes", "pem", "public", dsa_pub_pem_encoder_functions),
-    ENCODER("DSA", "yes", "pem", "parameters", dsa_param_pem_encoder_functions),
+            dsa_param_to_text_encoder_functions),
+    ENCODER("DSA", "yes", "der", "private",
+            dsa_priv_to_der_encoder_functions),
+    ENCODER("DSA", "yes", "der", "public",
+            dsa_pub_to_der_encoder_functions),
+    ENCODER("DSA", "yes", "der", "parameters",
+            dsa_param_to_der_encoder_functions),
+    ENCODER("DSA", "yes", "pem", "private",
+            dsa_priv_to_pem_encoder_functions),
+    ENCODER("DSA", "yes", "pem", "public",
+            dsa_pub_to_pem_encoder_functions),
+    ENCODER("DSA", "yes", "pem", "parameters",
+            dsa_param_to_pem_encoder_functions),
 #endif
 
 #ifndef OPENSSL_NO_EC
     ENCODER("X25519", "yes", "text", "private",
-        x25519_priv_print_encoder_functions),
+            x25519_priv_to_text_encoder_functions),
     ENCODER("X25519", "yes", "text", "public",
-        x25519_pub_print_encoder_functions),
+            x25519_pub_to_text_encoder_functions),
     ENCODER("X25519", "yes", "der", "private",
-        x25519_priv_der_encoder_functions),
-    ENCODER("X25519", "yes", "der", "public", x25519_pub_der_encoder_functions),
+            x25519_priv_to_der_encoder_functions),
+    ENCODER("X25519", "yes", "der", "public",
+            x25519_pub_to_der_encoder_functions),
     ENCODER("X25519", "yes", "pem", "private",
-        x25519_priv_pem_encoder_functions),
-    ENCODER("X25519", "yes", "pem", "public", x25519_pub_pem_encoder_functions),
+            x25519_priv_to_pem_encoder_functions),
+    ENCODER("X25519", "yes", "pem", "public",
+            x25519_pub_to_pem_encoder_functions),
 
-    ENCODER("X448", "no", "text", "private", x448_priv_print_encoder_functions),
-    ENCODER("X448", "no", "text", "public", x448_pub_print_encoder_functions),
-    ENCODER("X448", "no", "der", "private", x448_priv_der_encoder_functions),
-    ENCODER("X448", "no", "der", "public", x448_pub_der_encoder_functions),
-    ENCODER("X448", "no", "pem", "private", x448_priv_pem_encoder_functions),
-    ENCODER("X448", "no", "pem", "public", x448_pub_pem_encoder_functions),
+    ENCODER("X448", "no", "text", "private",
```

Comment:
Do you know why X448 is 'no'..
It is in the fips provider.. So I dont quite understand this..

Reply:
That happened in dfc0857d8191d43be320f4ba472b7c782248a35d, and I suppose it is because the previous setup of the encoder (then serializer) array didn't have "fips=yes" on these entries...  looking further back in history, that it seems that you were quite involved in making that decision, back in f552d900459cbacd82433b688b237cd6870663cd

Reply:
f552d90 is an EC related PR.. This is ecx related.

Reply:
And yet, there are changes to the ecx entries in there:

https://github.com/openssl/openssl/commit/f552d900459cbacd82433b688b237cd6870663cd#diff-56b17cde3b8bdbfa444192feb14efdc6L473-L485

Reply:
So no, I don't know.  I wasn't involved.

Reply:
Its a PR about ec so I dont know what that has to do with this..

Reply:
I will add another PR..

Reply:
Already fixed, no problem